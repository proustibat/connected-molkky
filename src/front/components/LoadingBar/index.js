import React from 'react';
import defaultStyle from './style';

const LoadingBar = () => (
  <div style={defaultStyle.container}>
    <div className="progress teal darken-4" style={defaultStyle.progressBar}>
      <div className="indeterminate" />
    </div>
  </div>
);

export default LoadingBar;
