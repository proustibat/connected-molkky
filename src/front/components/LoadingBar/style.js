export default {
  container: {
    background: '#fff',
    position: 'fixed',
    top: 0,
    left: 0,
    width: '100%',
    height: '2.2rem',
    zIndex: 10,
  },

  progressBar: {
    height: '1rem',
  },
};
