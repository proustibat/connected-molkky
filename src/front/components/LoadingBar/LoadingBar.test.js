import React from 'react';
import { shallow } from 'enzyme';
import LoadingBar from './index';

describe('LoadingBar', () => {
  it('should render the component correctly', () => {
    // Given / When
    const component = shallow(<LoadingBar />);

    // Then
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
