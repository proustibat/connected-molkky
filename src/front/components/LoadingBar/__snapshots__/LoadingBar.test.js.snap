// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingBar should render the component correctly 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "height": "2.2rem",
      "left": 0,
      "position": "fixed",
      "top": 0,
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <div
    className="progress teal darken-4"
    style={
      Object {
        "height": "1rem",
      }
    }
  >
    <div
      className="indeterminate"
    />
  </div>
</div>
`;
